Storage/SQL/SQLiteHistory.swift

```diff
@@ -997,7 +999,7 @@ extension SQLiteHistory: SyncableHistory {
         "v1.siteID AS siteID, v1.date AS visitDate, v1.type AS visitType " +
         "FROM " +
         "visits AS v1 " +
-        "JOIN history ON history.id = v1.siteID AND \(filter) " +
+        "JOIN (\(historyLimit)) as history ON history.id = v1.siteID AND \(filter) " +
```

Comment:
Get rid of `filter`, and inline the history part (`should_upload = 1`) inside the subquery — before the `LIMIT`! — and leave the visit test `v1.type <> 0` here.